### ISSUE TITLE
Need ability to specify tmpdir for remote

### ISSUE THREAD

**Issue by Endareth** [borgbackup/borg#4540]
## Have you checked borgbackup docs, FAQ, and open Github issues?
Yes

## Is this a BUG / ISSUE report or a QUESTION?
Issue

#### Your borg version (borg -V).
borg 1.1.5/borg 1.1.9

#### Operating system (distribution) and version.
Ubuntu 18.04 (local), QNAP 4.3.6.0923

#### Hardware / network configuration, and filesystems used.
Various, n/a to issue

#### How much data is handled by borg?
Various, n/a to issue

#### Full borg commandline that lead to the problem (leave away excludes and passwords)
Any use of remote repo

## Describe the problem you're observing.
Temp directory on remote full. The remote in question is a QNAP NAS, and configured with a tiny /tmp directory (64MB!).

#### Can you reproduce the problem? If so, describe how. If not, describe troubleshooting steps you took before opening the issue.
Running multiple backups results in multiple /tmp/_M* directories on remote. While some calls seem to cleanup after themselves, I just re-checked and saw multiple copies of that those _M tmp directories containing all the borg binaries/libraries.

#### Include any warning/errors/backtraces from the system logs
`Failed to write all bytes for _bisect.so`
From running a borg command remotely once the /tmp drive is full.

So to summarise, due to the QNAP NAS having a way too small /tmp directory (only 64MB allocated to it), being able to specify the tmp directory for a remote, either via env variable or command line option, would be a huge help.

**Comment by fantasya-pbem** [comment]
According to doc, BorgBackup should check temp dir env variables. See here:
https://docs.python.org/3/library/tempfile.html#tempfile.gettempdir

**Comment by ThomasWaldmann** [comment]
https://borgbackup.readthedocs.io/en/stable/usage/general.html#environment-variables

see last item there.

**Comment by ThomasWaldmann** [comment]
For a "borg serve" server, you need to have this in the environment of the "borg serve" process, of course.

**Comment by Endareth** [comment]
Thanks for the tips. Looks like QNAP doesn't set any of the temp variables in a standard sshd session. I was able to finally get it working by setting `PermitUserEnvironment yes` in the sshd_config, then adding `environment="TEMP=/root/tmp"` to the start of each of the SSH public key entries in authorized_hosts to use a different location. Rather clunky, but at least it works. Would still be nice to have an safer way of setting this for the borg remote, to avoid having to script changes to my sshd_config and authorized_keys file.

**Comment by ThomasWaldmann** [comment]
@Endareth can you make a PR adding this to the docs?

**Comment by ThomasWaldmann** [comment]
fixed by #4545.
